### DISCUSSION TITLE
Hiera lookup_options to match hash key in an array

### DISCUSSION THREAD

**Discussion by RobrVann** [puppetlabs/community#81] [Q&A]
Is there any way to match for a hash's key? I've tried using something akin to `'^some_module::(.*)2$':` but that only seems to match on the hiera key that contains the hash. For the example below, I'm looking to match `2` in the example below:

```
some_module::some_class::hash:
  key1:
    child_key1: something
    child_key2: something
    child_key3:
      - 1:   something
        2:   something_else
      - 1:   something
        2:   something_else
```

**Comment by kenyon** [comment]
Probably not possible. What are you trying to do in the `lookup_options`? I would try to write the class parameters in a way that you don't need to do this. If it's not a class that you can change, you could always write a wrapper class.

**Reply by RobrVann** [reply]
> What are you trying to do in the `lookup_options`?

Trying to convert the data type. I also tried converting the child_key3 array of hashes to just hashes to no avail.
> Probably not possible.

This is what I was expecting, but thought there might be something I don't know. I'll take care of it in the class. Thanks anyway.
